core/src/main/scala/io/projectglow/vcf/VariantContextToVCFRowConverter.scala

```diff
@@ -16,46 +16,38 @@
 
 package io.projectglow.vcf
 
-import java.lang.{Boolean => JBoolean, Iterable => JIterable}
-
-import scala.collection.mutable
+import java.lang.{Double => JDouble}
+import java.util.{List => JList}
 
 import htsjdk.samtools.ValidationStringency
 import htsjdk.variant.variantcontext.{VariantContext => HtsjdkVariantContext}
-import htsjdk.variant.vcf.{VCFConstants, VCFHeader}
+import htsjdk.variant.vcf.{VCFEncoderUtils, VCFHeader}
 
 import io.projectglow.common.{GlowLogging, VCFRow}
 
-// HTSJDK VariantContext -> VCFRow
-// Based on the HTSJDK classes VCFEncoder and CommonInfo
 object VariantContextToVCFRowConverter {
 
+  // Usually encodes the object normally, but does not pretty-print doubles
```

Comment:
My concern here is if we read a VCF and write it back again we should get exactly the same file (perhaps excluding the header). Does this change make sure that happens? It would be problematic for the used if for example they read a file containing doubles and write them back to disk and they get some other formatting of doubles. Your added test on parsedObjectAsString shows that things work correctly for that function. 

But I do not remember if our unit tests verify 100% conformance of read VCF and written one, e.g., whether their assert statements compare the verbatim strings from written file with the original file or they themselves do some formatting and hence mess up the comparison. Perhaps it is a good idea to read and write back an example VCF and compare visually to make sure things work correctly as this was not the case before https://github.com/projectglow/glow/commit/15e102e984a337fbd13a023d7c0e71dc582b7dcc

Reply:
These are tested via round-trips in the `VCFFileWriterSuite`s. In particular, the `Read single sample VCF with VCF parser` test compares the value of the `Double` field `ReadPosRankSum=1.075` (the usual VCF encoder prettifies this as `1.08`). `Read VEP VCF with VCF parser` compares the value of an `Array(Double)` field as well.